src/step-function.ts

```diff
@@ -56,32 +52,27 @@ export namespace $SFN {
    *
    * @see https://docs.aws.amazon.com/step-functions/latest/dg/amazon-states-language-wait-state.html
    */
-  // @ts-ignore
-  export function waitFor(seconds: number): void;
-
-  export function waitFor(call: CallExpr, context: CallContext): Wait {
-    if (context.kind !== ASL.ContextName) {
-      throw new Error(
-        `$SFN.wait is only available in the ${ASL.ContextName} context`
-      );
-    }
-    const seconds = call.args[0].expr;
-    if (seconds === undefined) {
-      throw new Error(`the 'seconds' argument is required`);
-    }
+  export const waitFor = makeIntegration<(seconds: number) => void>({
+    ...sfnIntegrationBase("waitFor"),
```

Comment:
This makes me wonder why aren't we using classes and inheritance? Seems overly complex? We've gone from a Function to now spreading a base object.

Reply:
```ts
export const waitFor = new BaseStepFunctionIntegration<(seconds: number) => void>({
  asl(call) { .. }
});
```

Reply:
or just a function, same result though. 

```ts
export const waitFor = makeStepFunctionIntegration<(seconds: number) => void>({
  asl(call) { .. }
});

const makeStepFunctionIntegration<F extends AnyFunction> = (integration: Omit<Integration, 'kind'>, operation: string) => makeIntegration<F>({
   kind: `SFN.${kind}`,
   ...integration
})
```